src/cascade/input_data/emr.py

```diff
@@ -0,0 +1,219 @@
+            # Integrate over age
+            divisor = prevalence_measurement.age_upper - prevalence_measurement.age_lower
+
+            csmr_mean = mean_interp["age"].integral(prevalence_measurement.age_lower, prevalence_measurement.age_upper)
+            csmr_mean /= divisor
+
+            csmr_stderr = stderr_interp["age"].integral(
+                prevalence_measurement.age_lower, prevalence_measurement.age_upper
+            )
+            csmr_stderr /= divisor
+        elif prevalence_measurement.age_lower == prevalence_measurement.age_upper:
+            # Integrate over time
+            divisor = prevalence_measurement.time_upper - prevalence_measurement.time_lower
+
+            csmr_mean = mean_interp["time"].integral(
+                prevalence_measurement.time_lower, prevalence_measurement.time_upper
+            )
+            csmr_mean /= divisor
+
+            csmr_stderr = stderr_interp["time"].integral(
+                prevalence_measurement.time_lower, prevalence_measurement.time_upper
+            )
+            csmr_stderr /= divisor
+        else:
+            # Integrate over both
+            divisor = (prevalence_measurement.time_upper - prevalence_measurement.time_lower) * (
+                prevalence_measurement.age_upper - prevalence_measurement.age_lower
+            )
+
+            csmr_mean = mean_interp["both"].integral(
+                prevalence_measurement.age_lower,
+                prevalence_measurement.age_upper,
+                prevalence_measurement.time_lower,
+                prevalence_measurement.time_upper,
+            )
+            csmr_mean /= divisor
+
+            csmr_stderr = stderr_interp["both"].integral(
+                prevalence_measurement.age_lower,
+                prevalence_measurement.age_upper,
+                prevalence_measurement.time_lower,
+                prevalence_measurement.time_upper,
+            )
+            csmr_stderr /= divisor
+
+        emr_mean = csmr_mean / prevalence_measurement["mean"]
+
+        # Propagation of uncertainty for f = A/B gives a standard deviation of
+        # sqrt( (sigma_a)^2 / A^2 + (sigma_b)^2 / B^2 - 2 (sigma_ab) / (A B) )
+        # but we don't know the covariance sigma_ab, so we use an upper bound on the error by
+        # assuming it's zero.
+        emr_stderr = np.sqrt(
+            (prevalence_measurement.standard_error / prevalence_measurement["mean"]) ** 2
+            + (csmr_stderr / csmr_mean) ** 2
+        )
+        return pd.Series({"mean": float(emr_mean), "standard_error": float(emr_stderr)})
+
+    return emr.merge(prevalence.apply(emr_mean, "columns"), left_index=True, right_index=True)
+
+
+def add_emr_from_prevalence(model_context, execution_context):
+    r"""Estimate excess mortality from the supplied prevalence measurements and
+    cause specific mortality estimates from GBD using the formula:
+    :math:`\chi=\frac{P}{{}_nm_x^c}
```

Comment:
I am trying to read this, and I am having trouble...

Reply:
Complain at Drew. He wrote it. It should render real nice in the Sphinx docs.